src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java

```diff
@@ -1056,5 +1057,31 @@ private boolean getMoreAvailable(Page<Organisation> pageableOrganisations) {
         return !pageableOrganisations.isLast();
     }
 
+    @Override
+    public ResponseEntity<ContactInformationResponse> updateContactInformationForOrganisation(
+        ContactInformationCreationRequest contactInfo, String organisationIdentifier) {
+
+        Organisation organisation = organisationRepository.findByOrganisationIdentifier(organisationIdentifier);
+        List<ContactInformation> existingContactInformationList = organisation.getContactInformation();
+        //if multiple contact informations are found they will all be updated
```

Comment:
1. If single address is present then update address details
2. If multiple addresses present then update address details for provided UPRN (investigate live DB has UPRN values)

Reply:
added condition to check if single address then update , if multiple addresses then check uprn an dupdate those records however if uprn not present a resource not found is thrown